asyncio_mqtt/client.py

```diff
@@ -380,7 +382,7 @@ def _pending_calls(self) -> Generator[int, None, None]:
         yield from self._pending_unsubscribes.keys()
         yield from self._pending_publishes.keys()
 
-    async def connect(self, *, timeout: int = 10) -> None:
+    async def connect(self, *, timeout: int | float | None = None) -> None:
```

Comment:
Good idea to have method-local overrides to `timeout`. :+1: 

I propose that we use `timeout: float | None = None` (see the previous comment).
Furthermore, I propose the following semantics (written in psedo-code here):

```python
if timeout is None:
    "Use the instance default (`self._timeout`)"
else:
    "Use the `timeout` argument"
```